src/ng/rootScope.js

```diff
@@ -786,12 +786,12 @@ function $RootScopeProvider() {
           current = target;
 
           // It's safe for asyncQueuePosition to be a local variable here because this loop can't
-          // be reentered recursively. Calling $digest from a function passed to $applyAsync would
+          // be reentered recursively. Calling $digest from a function passed to $evalAsync would
           // lead to a '$digest already in progress' error.
           for (var asyncQueuePosition = 0; asyncQueuePosition < asyncQueue.length; asyncQueuePosition++) {
             try {
               asyncTask = asyncQueue[asyncQueuePosition];
-              asyncTask.scope.$eval(asyncTask.expression, asyncTask.locals);
+              asyncTask.expression(asyncTask.scope, asyncTask.locals);
```

Comment:
Yes, I mean functions. I think it does (matter). For example, having a function that incorrectly tries to access `this.someProp`, would previously throw (because `this === undefined`), but will now silently fail. 

I would rather not have the current behavior changed.